### DISCUSSION TITLE
Adding source-watcher to source controller: no kind is registered for the type v1beta1.Gitrepository in the scheme

### DISCUSSION THREAD

**Discussion by peldenwong** [fluxcd/flux2#3055] [Q&A] [answered]
I am trying to add the source-watcher to the source controller. I followed the guide, the `init()` function is already initialised correctly as follow:
```
func init() {
	utilruntime.Must(clientgoscheme.AddToScheme(scheme))

	utilruntime.Must(sourcev1.AddToScheme(scheme))
	// +kubebuilder:scaffold:scheme
}
```
and I've added the following to the main function:

```
if err = (&controllers.GitRepositoryWatcher{
		Client: mgr.GetClient(),
		Scheme: mgr.GetScheme(),
	}).SetupWithManager(mgr); err != nil {
		setupLog.Error(err, "unable to create controller", "controller", "GitRepositoryWatcher")
		os.Exit(1)
	}
```

I then built and used the updated image. Attached is the error I'm getting while trying to install the updated controller.
<img width="1493" alt="failed controller" src="https://user-images.githubusercontent.com/77985102/187478226-7ab8e932-71ea-4b7d-a182-7074f7b7fd4d.png">

Any help is appreciated. Thank you!

**Comment by stefanprodan** [comment, marked as answer]
Have you installed Flux on your cluster? See https://fluxcd.io/flux/gitops-toolkit/source-watcher/#install-flux

**Reply by peldenwong** [reply]
flux is installed. Below is the output of `flux check --pre`:

```
checking prerequisites
flux 0.31.4 <0.33.0 (new version is available, please upgrade)
Kubernetes 1.21.5 >=1.20.6-0
prerequisites checks passed
```

**Reply by stefanprodan** [reply]
If you clone this repo and and you run it locally, does it error the same? https://github.com/fluxcd/source-watcher

**Reply by peldenwong** [reply]
I found the problem. The controller in https://github.com/fluxcd/source-watcher imports `github.com/fluxcd/source-controller/api/v1beta1`. Importing to `github.com/fluxcd/source-controller/api/v1beta2` solves the issue above, however, creating a GitRepository now gives the following error `no matches for kind "GitRepository" in version "source.toolkit.fluxcd.io/v1beta2"`.

**Reply by stefanprodan** [reply]
Can you please try using latest Flux 0.33. It should work fine with v1beta2.
